PoseEstimation/poseEstimation_main.py

```diff
@@ -0,0 +1,171 @@
+
+import deeplabcut
+print("Imported DLC!", flush=True)
+
+# Import for PyInstaller
+from multiprocessing import freeze_support
+freeze_support()
+
+def main(poseEstimation_params: poseEst_params.PoseEstimationParameters):
+
+    """
+    DeepLabCut algorithm
+    """
+    # --------------- Read danse parameters ---------------
+    operations: str      = poseEstimation_params.params[defs.DoricFile.Attribute.Group.OPERATIONS]
+    filepath: str        = poseEstimation_params.paths[defs.Parameters.Path.FILEPATH]
+    datapath: str        = poseEstimation_params.paths[defs.Parameters.Path.H5PATH]
+    project_folder: str  = poseEstimation_params.params[poseEst_defs.Parameters.danse.PROJECT_FOLDER]
+    bodypart_names       = poseEstimation_params.params[poseEst_defs.Parameters.danse.BODY_PART_NAMES].split(', ')
+    bodypart_colors      = poseEstimation_params.params[poseEst_defs.Parameters.danse.BODY_PART_COLORS].split(', ')
+    extracted_frames     = poseEstimation_params.params[poseEst_defs.Parameters.danse.EXTRACTED_FRAMES]
+
+    [file_, video_path, path_config_file, training_coordinates] = create_project(filepath, datapath, project_folder, bodypart_names, extracted_frames, poseEstimation_params)
+    deeplabcut.create_training_dataset(path_config_file)
+    deeplabcut.train_network(path_config_file)
+    deeplabcut.evaluate_network(path_config_file)
+ 
+    path_output = path_config_file.rsplit("\\", 1)[0]
+    deeplabcut.analyze_videos(path_config_file, [video_path], destfolder = path_output)
+
+    save_coords_to_doric(file_, datapath, path_output, extracted_frames, bodypart_names, project_folder, bodypart_colors, operations, 
+                         training_coordinates, groupNames = poseEstimation_params.get_h5path_names())
+
+def preview(poseEstimation_params: poseEst_params.PoseEstimationParameters):
+    print("hello preview")
+
+def create_project(filepath, datapath, project_folder, bodypart_names, extracted_frames, poseEstimation_params):
+    task         = os.path.splitext(os.path.basename(filepath))[0] 
+    experimenter = "danse"
+    file_        = h5py.File(filepath, 'a')
+    attributes   = utils.load_attributes(file_, datapath)
+    relativePath = attributes[poseEst_defs.Parameters.danse.RELATIVE_FILEPATH]
+    dir          = os.path.dirname(filepath)
+    video_path   = os.path.join(dir, relativePath)
+    video_path   = video_path.replace("\\", "/")
+
+    path_config_file: str = deeplabcut.create_new_project(task, experimenter, [video_path], project_folder, copy_videos = False)
+    update_config_file(path_config_file, bodypart_names)
+
+    training_coordinates = create_labeled_data_HDF(path_config_file, extracted_frames, bodypart_names, experimenter, poseEstimation_params, video_path)
+
+    return [file_, video_path, path_config_file, training_coordinates]
+
+def create_labeled_data_HDF(path_config_file, extracted_frames, bodypart_names, experimenter, poseEstimation_params, video_path):
```

Comment:
Just `create_labeled_data`
Please add comments to this function to facilitate the reading